operator/cmd/cilium_node.go

```diff
@@ -269,8 +274,18 @@ func syncHandlerConstructor(notFoundHandler func(name string), foundHandler func
 		}
 		cn, ok := obj.(*cilium_v2.CiliumNode)
 		if !ok {
-			log.Errorf("Object stored in store is not *cilium_v2.CiliumNode but %T", obj)
-			return err
+			tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				return fmt.Errorf("couldn't get object from tombstone %#v", obj)
+			}
+			cn, ok = tombstone.Obj.(*cilium_v2.CiliumNode)
+			if !ok {
+				return fmt.Errorf("tombstone contained object that is not a *cilium_v2.CiliumNode %#v", obj)
+			}
```

Comment:
I don't think we should be splatting the obj into the error msg. We only care about the type, right?

```suggestion
				return fmt.Errorf("couldn't get object from tombstone %T", obj)
			}
			cn, ok = tombstone.Obj.(*cilium_v2.CiliumNode)
			if !ok {
				return fmt.Errorf("tombstone contained object that is not a *cilium_v2.CiliumNode %T", obj)
			}
```